apps/teca_temporal_reduction.cpp

```diff
@@ -0,0 +1,269 @@
+        ("full_help", "\ndisplays both basic and advanced documentation together\n")
+        ;
+
+    options_description advanced_opt_defs(
+        "Advanced usage:\n\n"
+        "The following list contains the full set options giving one full\n"
+        "control over all runtime modifiable parameters. The basic options\n"
+        "(see" "--help) map to these, and will override them if both are\n"
+        "specified.\n\n"
+        "Advanced command line options", help_width, help_width - 4
+        );
+
+    p_teca_multi_cf_reader mcf_reader = teca_multi_cf_reader::New();
+    p_teca_cf_reader cf_reader = teca_cf_reader::New();
+    p_teca_valid_value_mask vv_mask = teca_valid_value_mask::New();
+    p_teca_unpack_data unpack = teca_unpack_data::New();
+    p_teca_temporal_reduction mav = teca_temporal_reduction::New();
+    p_teca_cf_writer cf_writer = teca_cf_writer::New();
+
+    cf_reader->get_properties_description("cf_reader", advanced_opt_defs);
+    mcf_reader->get_properties_description("mcf_reader", advanced_opt_defs);
+    vv_mask->get_properties_description("vv_mask", advanced_opt_defs);
+    unpack->get_properties_description("unpack", advanced_opt_defs);
+    mav->get_properties_description("mav", advanced_opt_defs);
+    cf_writer->get_properties_description("cf_writer", advanced_opt_defs);
+
+    // package basic and advanced options for display
+    options_description all_opt_defs(help_width, help_width - 4);
+    all_opt_defs.add(basic_opt_defs).add(advanced_opt_defs);
+
+    // parse the command line
+    int ierr = 0;
+    variables_map opt_vals;
+    if ((ierr = teca_app_util::process_command_line_help(
+        mpi_man.get_comm_rank(), argc, argv, basic_opt_defs,
+        advanced_opt_defs, all_opt_defs, opt_vals)))
+    {
+        if (ierr == 1)
+            return 0;
+        return -1;
+    }
+
+    if (mpi_man.get_comm_rank() == 0 && opt_vals["verbose"].as<int>())
+    {
+       std::cerr << "running on " << mpi_man.get_comm_size() << " ranks" << std::endl;
+       std::cerr << "file_layout=" << opt_vals["file_layout"].as<string>() << std::endl;
+       std::cerr << "steps_per_file=" << opt_vals["steps_per_file"].as<long>() << std::endl;
+       std::cerr << "interval=" << opt_vals["interval"].as<string>() << std::endl;
+       std::cerr << "operator=" << opt_vals["operator"].as<string>() << std::endl;
+       std::cerr << "point_arrays=" << opt_vals["point_arrays"].as<std::vector<std::string>>()<< std::endl;
+       std::cerr << "ignore_fill_value=" << opt_vals["ignore_fill_value"].as<int>() << std::endl;
+       std::cerr << "fill_value=" << opt_vals["fill_value"].as<double>() << std::endl;
+    }
+
+    mcf_reader->set_properties("mcf_reader", opt_vals);
+    cf_reader->set_properties("cf_reader", opt_vals);
+    vv_mask->set_properties("vv_mask", opt_vals);
+    unpack->set_properties("unpack", opt_vals);
+    mav->set_properties("mav", opt_vals);
```

Comment:
Use a more descriptive name here because this is displayed in the help message and used to set advanced property values.